main/internal/implementations/terraform_workspace/azure_backend.go

```diff
@@ -0,0 +1,113 @@
+package terraformWorkspace
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+
+	terraformValueObjects "github.com/dragondrop-cloud/cloud-concierge/main/internal/implementations/terraform_value_objects"
+	"github.com/dragondrop-cloud/cloud-concierge/main/internal/interfaces"
+)
+
+// AzureBlobBackend is an implementation of the interfaces.TerraformWorkspace interface that uses Azure Blob Storage as
+type AzureBlobBackend struct {
+	// config is the configuration for the Azure Blob Storage backend.
+	config ContainerBackendConfig
+
+	// dragonDrop is the DragonDrop interface that is used to communicate with the DragonDrop API.
+	dragonDrop interfaces.DragonDrop
+
+	// workspaceToBackendDetails is a map of Terraform workspace names to their respective backend details.
+	workspaceToBackendDetails map[string]interface{}
+}
+
+// FindTerraformWorkspaces returns a map of Terraform workspace names to their respective directories.
+func (b *AzureBlobBackend) FindTerraformWorkspaces(ctx context.Context) (map[string]string, error) {
+	workspaces, workspaceToBackendDetails, err := findTerraformWorkspaces(ctx, b.dragonDrop, b.config.WorkspaceDirectories, "azure_blob")
+	if err != nil {
+		return nil, err
+	}
+	b.workspaceToBackendDetails = workspaceToBackendDetails
+
+	return workspaces, err
+}
+
+// NewAzureBlobBackend creates a new AzureBlobBackend instance.
+func NewAzureBlobBackend(ctx context.Context, config ContainerBackendConfig, dragonDrop interfaces.DragonDrop) interfaces.TerraformWorkspace {
```

Comment:
```suggestion
func NewAzurermBlobBackend(ctx context.Context, config ContainerBackendConfig, dragonDrop interfaces.DragonDrop) interfaces.TerraformWorkspace {
```
To keep consistent with terraform naming https://developer.hashicorp.com/terraform/language/settings/backends/azurerm